p2p/module.go

```diff
@@ -199,3 +217,99 @@ func (m *p2pModule) handleNetworkMessage(networkMsgData []byte) {
 
 	m.GetBus().PublishEventToBus(&event)
 }
+
+func configureBootstrapNodes(p2pCfg *configs.P2PConfig, m *p2pModule) error {
+	var csvReader *csv.Reader
+	customBootstrapNodesCsv := strings.Trim(p2pCfg.BootstrapNodesCsv, " ")
+	if customBootstrapNodesCsv == "" {
+		csvReader = csv.NewReader(strings.NewReader(defaults.DefaultP2PBootstrapNodesCsv))
+	} else {
+		csvReader = csv.NewReader(strings.NewReader(customBootstrapNodesCsv))
+	}
+	bootStrapNodes, err := csvReader.Read()
+	if err != nil {
+		return fmt.Errorf("error parsing bootstrap nodes: %w", err)
+	}
+	for _, node := range bootStrapNodes {
+		if !isValidHostnamePort(node) {
+			return fmt.Errorf("invalid bootstrap node: %s", node)
+		}
+	}
+	m.bootstrapNodes = bootStrapNodes
+	return nil
+}
+
+func isValidHostnamePort(str string) bool {
+	pattern := regexp.MustCompile(`^(https?)://([a-zA-Z0-9.-]+):(\d{1,5})$`)
+	matches := pattern.FindStringSubmatch(str)
+	if len(matches) != 4 {
```

Comment:
```suggestion
	if matches != nil || len(matches) != 4 {
```

Reply:
I think it's redundant since the `nil` case in included in "not len(4)"

```go
nill := []string(nil)
if len(nill) != 4 {
    fmt.Println("not 4")
}
```
Prints `not 4`